textmap/tokenizers.py

```diff
@@ -182,129 +131,93 @@ def fit(self, X, **fit_params):
         -------
         self
         """
-        if self.tokenize_by == "sentence":
-            if self.lower_case:
-                self.tokenization_ = [
-                    [word_tokenize(sent.lower()) for sent in sent_tokenize(doc)]
-                    for doc in X
-                ]
-            else:
-                self.tokenization_ = [
-                    [word_tokenize(sent) for sent in sent_tokenize(doc)] for doc in X
-                ]
+
+        if self.lower_case:
+            tokenize = lambda d: [w.lower() for w in self.nlp.tokenize(d)]
         else:
-            if self.lower_case:
-                self.tokenization_ = [word_tokenize(doc.lower()) for doc in X]
-            else:
-                self.tokenization_ = [word_tokenize(doc) for doc in X]
+            tokenize = lambda d: self.nlp.tokenize(d)
 
-        self.iteratively_contract_bigrams()
+        if self.tokenize_by == "sentence":
+            self.tokenization_ = flatten(
+                [[tokenize(sent) for sent in sent_tokenize(doc)] for doc in X]
+            )
+        else:
+            self.tokenization_ = [tokenize(doc) for doc in X]
```

Comment:
You may want to make this an
```suggestion
        elif self.tokenize_by == "document":
            self.tokenization_ = [tokenize(doc) for doc in X]
        else:
            raise ValueError(f"Unrecognized tokenize_by argument {self.tokenize_by}. Should be one of \"sentence\" or \"document\"")
```

Reply:
Cool.  I can do that.  There were a few typo's that I'm fixing up as well.